doc/pages/development-guide/001-dev_guide.md

```diff
@@ -0,0 +1,165 @@
+Refer to the following [”Testing a single component”](#testing-a-single-component) paragraph for dependencies, then you can start your application with an interactive shell by replacing `mix test` with `iex -S mix`
+
+## Ok, I made changes, what now?
+
+Once you got your hands dirty, is time to test them.
+You can test changes to a single component, or run E2E tests if more than one service is involved.
+
+Remember that a contribution should always include tests for the new functionality or the fix.
+We use the [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) framework for unit testing.
+Work is under way to add property-based testing too! 
+
+### Testing a single component
+
+In general, you will have to bring up a RabbitMQ and a Scylla instance (not all services need both: API services will not need access to the database).
+You can do so by running
+
+```bash
+docker run --rm -p 5672:5672 -p 15672:15672 rabbitmq:management
+docker run --rm -p 9042:9042 scylladb/scylla
+```
+
+Then, you can run test in the component directory with
+
+```bash
+RABBITMQ_HOST=localhost CASSANDRA_NODES=localhost mix test --exclude wip
+```
+
+> [!NOTE]
+> Some services are a bit special (for now!) and might need a little more setup for testing.
+> - AppEngine API needs the AMQP exchange `astarte_events` to be declared:
+> after having started RabbitMQ, run the following lines to declare it:
+>  ```bash
+>   wget http://guest:guest@localhost:15672/cli/rabbitmqadmin -O rabbitmqadmin
+>   chmod +x ./rabbitmqadmin
+>   ./rabbitmqadmin declare exchange name=astarte_events type=direct
+>   rm rabbitmqadmin
+>  ```
+
+> - Pairing needs a CFSSL instance available and exposed on port 8080: you can run
+>  ```bash
+>  docker run --net=host -p 8080/tcp ispirata/docker-alpine-cfssl-autotest:astarte
+>  ```
+> 
+> and then test using
+>  ```bash
+>  RABBITMQ_HOST=localhost CASSANDRA_NODES=localhost CFSSL_API_URL=http://localhost:8080 mix test
+>  ```
+
+### E2E testing
+
+docker compose is the quick-and-dirty way to spin up a development instance of Astarte.
+You can do so just by running `docker compose up -d` in the main Astarte directory.
+You can rebuild the service you’re working on with `docker compose build $SERVICE_NAME`,
+or edit the `docker-compose.yml` file to change the service image name.
+
+ Some useful commands:
+
+- `docker compose up -d` starts all containers in detached mode
+- `docker compose down $SERVICE_NAME -v` stops and cleans the service you’re working on
+- `docker compose up $SERVICE_NAME --build` rebuilds the service and restarts it. If you don’t need to tail logs, you may add `-d`
+
+## Useful commands
+
+Now, some commands you might find helpful to set up a realm:
+
+- Create a realm keypair:
+`astartectl utils gen-keypair $REALM`
+- Create a Realm:
+`astartectl housekeeping realms create $REALM -u http://api.astarte.localhost --housekeeping-key ./compose/astarte-keys/housekeeping_private.pem --realm-private-key $PRIVATE_PEM`
```

Comment:
user only have ./compose/astarte-keys/housekeeping_private.pem if they have followed astarte in 5 minutes before, better state that?